pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -208,8 +208,11 @@ func (ssc *defaultStatefulSetControl) getStatefulSetRevisions(
 			return nil, nil, err
 		}
 	} else {
+		if set.Status.CollisionCount == nil {
+			set.Status.CollisionCount = new(int64)
```

Comment:
Agree, as above, just add a comment to that effect for future contributors please.